src/player.cpp

```diff
@@ -10427,6 +10428,11 @@ void player::rooted()
 
 bool player::can_wield( const item &it, bool interactive ) const
 {
+    if( it.is_null() ) {
```

Comment:
> While it would need to be special cased in wield (we always want to wield our "personal" null item)

We only ever check `target.is_null()` and never actually assign it so the player-specific version is always assigned in that case. I've implemented the other changes in 49d809d